File: HrSystemFrontend/HrSystemFront/src/app/core/models/leave-policy.ts
export interface LeavePolicyResponse {
id: string;
  leaveType: string | number;
  isEnabled: boolean;
  annualAllowance: number;
  maxConsecutiveDays: number;
  minNoticeDays: number;
  backdateDays: number;
  version: number;
}

export interface CreateLeavePolicyRequest {
  leaveType: string;
  annualAllowance: number;
  maxConsecutiveDays: number;
  minNoticeDays: number;
  backdateDays: number;
}

export interface UpdateLeavePolicyRequest {
  id: string;
  annualAllowance: number;
  maxConsecutiveDays: number;
  minNoticeDays: number;
  backdateDays: number;
}

export interface UpdatePolicyStatusRequest {
  id: string;
  isEnabled: boolean;
}

File: HrSystem/HrSystem.Domain/Entities/LeavePolicy.cs
using HrSystem.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HrSystem.Domain.Entities
{
    public class LeavePolicy
    {
        public Guid Id { get; set; }
        public LeaveType LeaveType { get; set; }
        public bool IsEnabled { get; set; } = true;
        public int AnnualAllowance { get; set; }
        public int MaxConsecutiveDays { get; set; }
        public int MinNoticeDays { get; set; }
        public int BackdateDays { get; set; } // Specific to Sick Leave
        public int Version { get; set; } = 1;
    }
}
